new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -0,0 +1,1237 @@
+
+    patch = new V1Patch(new String(patchStr));
+    crdPatched = false;
+    crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(adminonly) failed");
+
+    assertTrue(crdPatched, "patchDomainCustomResource(adminonly) failed");
+    logger.info("!!! Domain Resource patched for shutdown (ADMIN_ONLY) !!!");
+
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+    checkServiceExists(adminServerPodName, domainNamespace);
+
+    // make sure all other managed server pods are not provisioned 
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodDeleted(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+    checkPodDeleted(configServerPodName, domainUid, domainNamespace);
+    checkPodDeleted(standaloneServerPodName, domainUid, domainNamespace);
+
+    logger.info("!!! Domain restart (ADMIN_ONLY) success !!!");
+
+    // Patch the Domain with serverStartPolicy set to IF_NEEDED
+    // Here all the Servers should come up
+    patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/serverStartPolicy\",")
+        .append(" \"value\":  \"IF_NEEDED\"")
+        .append(" }]");
+    logger.info("Domain restart(if_needed) patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    crdPatched = false;
+    crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(if_needed) failed");
+    assertTrue(crdPatched, "patchDomainCustomResource(if_needed) failed");
+    logger.info("!!! Domain Resource patched for shutdown (IF_NEEDED) !!!");
+    
+    // check dynamic managed server pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReady(managedServerPrefix + i, domainUid, domainNamespace);
+      checkServiceExists(managedServerPrefix + i, domainNamespace);
+    }
+    checkPodReady(configServerPodName, domainUid, domainNamespace);
+    checkPodReady(standaloneServerPodName, domainUid, domainNamespace);
+    logger.info("!!! Domain restart (IF_NEEDED) success !!!");
+  }
+
+  /**
+   * Add a second managed server (config-cluster-server2) in a configured 
+   * cluster with SeeverStartuoPolicy IF_NEEDED. 
+   * Initially, server will not come up since the replica count is set to 1.
+   * Update the serverStartPolicy for config-cluster-server2 to ALWAYS
+   * by patching the resource defintion with 
+   *  spec/managedServers/1/serverStartPolicy set to ALWAYS.
+   * Make sure that managed server config-cluster-server2 is up and running
+   * Stop the managed server by patching the resource defintion 
+   *   with spec/managedServers/1/serverStartPolicy set to IF_NEEDED.
+   * Make sure the specified managed server is stopped as per replica count.
```

Comment:
why would the managed server be stopped? When the replicaCount is 1 how does changing the serverStartPolicy from ALWAYS to IF_NEEDED shutsdown the server? There is only one running server in this cluster right?

Reply:
ALWAYS for managed server just start the managed server irrespective of replica count. You may go beyond the replica count. IF_NEEDED means it starts if we need to fulfill the replica count.
Ref : https://oracle.github.io/weblogic-kubernetes-operator/userguide/managing-domains/domain-lifecycle/startup/